### DISCUSSION TITLE
Feature Request: Remove all HTML Comments

### DISCUSSION THREAD

**Discussion by bacoords** [holger1411/understrap#2082] [Ideas]
Opening a new discussion to carry on what was discussed here: https://github.com/understrap/understrap/pull/1864

The topic: removing all HTML comments from Understrap. I think the PHP comments are valuable (and there is a lot of improvement on that front, especially moving comments for PHP functions out of HTML and into PHP) but there's a number of HTML comments that honestly we could do without. 

## Examples of unnecessary HTML Comments

Area comments 

```html
<!-- ******************* The Navbar Area ******************* -->
```

Closing tag comments

```html

	</div><!-- #content -->

</div><!-- #single-wrapper -->
```

## Request

So this is essentially a public poll: Are there strong arguments in favor of keeping the HTML comments in the template files or is it time to move past them?

**Comment by shikkaba** [comment]
Please don't do this. When I first started out with this theme, the comments were a godsend.

**Comment by IanDelMar** [comment]
> Please don't do this. When I first started out with this theme, the comments were a godsend.

May I ask in what way the comments were useful? And what comments exactly were useful?

**Comment by shikkaba** [comment]
Area comments. Helped with figuring out what was what.

**Comment by gettonet** [comment]
I always use this, might be handy for someone:
// Remove html comments from frontend.
`function cleanIt($buffer)
{
    $buffer = preg_replace('//', '', $buffer);
    return $buffer;
}
function buffer_start()
{
    ob_start("cleanIt");
}
function buffer_end()
{
    ob_end_flush();
}

add_action('get_header', 'buffer_start');
add_action('wp_footer', 'buffer_end');`

(and I also vote for removing them)

**Comment by IanDelMar** [comment]
@gettonet I recommend to not use this code. Your're literally matching nothing. If you want to remove `//` (which indicates a PHP comment and would only remove the `//` part, making the comment visible to users or even cause a fatal error) you would have to use `$buffer = preg_replace('/\/\//', '', $buffer);`. Also, this removes `//` from all string, eg. `https://`.

Here's what it should look like:
```php
$buffer = preg_replace('/<!--(.*)-->/U', '', $buffer); // removes <!-- <p>I am a comment!</p> -->
```

**Comment by gettonet** [comment]
@IanDelMar I just saw that code was sanitized and is not showing properly in comment :)
You're absolutely right, the code I use is actually (let's try again):

`$buffer = preg_replace('/<!--(.|s)*?-->/', '', $buffer);`

**Comment by IanDelMar** [comment]
@bacoords is it possible to move this issue to the discussions page and have a poll over there?

![image](https://user-images.githubusercontent.com/42134098/215800658-6a138459-8d8f-4d13-8ddf-fea92acc268c.png)

**Reply by bacoords** [reply]
Yes good idea.

**Comment by pattonwebz** [comment]
As a matter of personal preference (talking purely about the 2 types of HTML comments above) I have no need or never had value from seeing large section comments in any markup. 

I do however find great value in ending comments for any tags which may open in one file but close in another. A common example is the content area opening in the `header.php` file and closing in the `footer.php` file.

No strong opinions on them by any means. Just adding one other voice and thoughts to the discussion :)
